sql/hive/src/test/scala/org/apache/spark/sql/hive/PartitionedTablePerfStatsSuite.scala

```diff
@@ -405,8 +405,8 @@ class PartitionedTablePerfStatsSuite
           })
           executorPool.shutdown()
           executorPool.awaitTermination(30, TimeUnit.SECONDS)
-          assert(HiveCatalogMetrics.METRIC_FILES_DISCOVERED.getCount() == 50)
-          assert(HiveCatalogMetrics.METRIC_PARALLEL_LISTING_JOB_COUNT.getCount() == 1)
+          assert(HiveCatalogMetrics.METRIC_FILES_DISCOVERED.getCount() == 100)
+          assert(HiveCatalogMetrics.METRIC_PARALLEL_LISTING_JOB_COUNT.getCount() == 2)
```

Comment:
The problem is in https://github.com/apache/spark/blob/3fdbc48373cdf12b8ba05632bc65ad49b7af1afb/sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala#L517
It triggers additional files listen. It seems we triggered that everywhere where we replaced `sessionCatalog.refreshTable` by `CatalogImpl.refreshTable()` or added a call to `CatalogImpl.refreshTable()`.

cc @cloud-fan @HyukjinKwon @sunchao @dongjoon-hyun

Reply:
can we make it clear that when shall we call `sessionCatalog.refreshTable` and when shall we call `CatalogImpl.refreshTable()`?

Reply:
I think there are following cases for `SessionCatalog. refreshTable`:
1. Modification in a table that don't affect table data. For example, updating table stats
2. Table data cache is updated/handled somehow separately like `DROP TABLE` drop the cache or `ALTER TABLE table1 RENAME TO table2` re-creates the cache itself

Other cases are for `CatalogImpl.refreshTable()`, it seems.